src/sensitivities/linalg/factorization/cholesky.jl

```diff
@@ -0,0 +1,198 @@
+import Base.LinAlg.BLAS: gemv, gemv!, gemm!, trsm!, axpy!, ger!
+import Base.LinAlg.chol
+
+#=
+See [1] for implementation details: pages 5-9 in particular. The derivations presented in
+[1] assume column-major layout, whereas Julia primarily uses row-major. We therefore
+implement both the derivations in [1] and their transpose, which is more appropriate to
+Julia.
+
+[1] - "Differentiation of the Cholesky decomposition", Murray 2016
+=#
+
+const AM = AbstractMatrix
+const UT = UpperTriangular
+@explicit_intercepts chol Tuple{AbstractMatrix{<:∇Real}}
+∇(::typeof(chol), ::Type{Arg{1}}, p, U::UT{T}, Ū::UT{T}, Σ::AM{T}) where T<:∇Real =
+    chol_blocked_rev(full(Ū), full(U), 25, 'U')
+
+"""
+    level2partition(A::AbstractMatrix, j::Int, ul::Char)
+
+Returns views to various bits of the lower triangle of `A` according to the
+`level2partition` procedure defined in [1] if `ul` is `'L'`. If `ul` is `'U'` then the
+transposed views are returned from the upper triangle of `A`.
+"""
+function level2partition(A::AM, j::Int, ul::Char)
```

Comment:
Will `level2partition` ever accept a value for `ul` that is different than `U` and `L`? If not, then it might make sense to change `ul::Char` to `lower::Bool` or `upper:Bool`.

Reply:
Good point. I was following the BLAS / LAPACK convention of using Chars for flags, which is pretty dumb. A Boolean would make much more sense.